cpp/include/cudf/io/types.hpp

```diff
@@ -147,6 +165,9 @@ struct sink_info {
 
   explicit sink_info(std::vector<char>* buffer)
       : type(io_type::HOST_BUFFER), buffer(buffer) {}
+
+  explicit sink_info()
```

Comment:
Build error due to two sink_info() constructors: we can probably just use the default constructor and change the default value of type to be io_type::VOID instead of io_type::FILEPATH.

Reply:
Yeah, I'm about to submit a PR update.